### DISCUSSION TITLE
Theme for consistency

### DISCUSSION THREAD

**Discussion by ozdotcom** [CarlosPietrasanta/Tafe230130#6] [Q&A]
@Jasonlovedataanalyst , @CarlosPietrasanta 

What color theme are you guys using.
are we keeping with the same consistent look over the 3 calculators?
I have used the following but happy to change to suit everyone.

I am having trouble with the focus color on buttons almost disappearing with this theme.
I tried adding this to the background but the theme still overrides it - FocusVisualPrimaryBrush="#FFF5DEDE"

`Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"  >
    <Page.Resources>
        <Style TargetType="TextBox">
            <Setter Property="Background" Value="#FFF3F3FB" />
            <Setter Property="Foreground" Value="DarkBlue" />
            <Setter Property="Margin" Value="10" />
            <Setter Property="MinWidth" Value="200" />
            <Setter Property="BorderBrush" Value="DarkBlue"/>
        </Style>
        <Style TargetType="TextBlock">
            <Setter Property="Foreground" Value="#FFF3F3FB" />
            <Setter Property="Margin" Value="10" />
            <Setter Property="MinWidth" Value="100" />
        </Style>
        <Style TargetType="Button">
            <Setter Property="Background" Value="#FFF3F3FB" />
            <Setter Property="Foreground" Value="DarkBlue" />
            <Setter Property="Padding" Value="10" />
            <Setter Property="Margin" Value="20" />
            <Setter Property="MinWidth" Value="120" />
            <Setter Property="BorderBrush" Value="#FFF3F3FB"/>
        </Style>
    </Page.Resources>
    <Grid Background="#FF0F0F0F"  >`

**Comment by CarlosPietrasanta** [comment]
That's an excellent idea. I was using the default theme but to add some colour we can definitely use the same theme to keep it consistent.

I tried using that theme and although I like it, I'm having the same issue. As far as I can remember, the only way to change the colour when the mouse is over the button is programatically, using C#. And personally I'd try to don't worry too much about that and keep the theme-related stuff it in XAML alone.

I modified that theme with lighter colours and looks like this:

![Screenshot 2023-03-23 193514](https://user-images.githubusercontent.com/127190272/227154823-e99791d0-3b34-471c-aeb3-ae7f4398365b.png)

And here it is to copy and paste into your XAML code. Of course feel free to suggest improvements.

	Background="{ThemeResource ApplicationPageBackgroundThemeBrush}" >
	<Page.Resources>
		<Style TargetType="TextBox">
			<Setter Property="Background" Value="LightCyan" />
			<Setter Property="Foreground" Value="DarkBlue" />
			<Setter Property="Margin" Value="10" />
			<Setter Property="MinWidth" Value="200" />
			<Setter Property="BorderBrush" Value="DarkBlue"/>
		</Style>
		<Style TargetType="TextBlock">
			<Setter Property="Foreground" Value="DarkBlue" />
			<Setter Property="Margin" Value="10" />
			<Setter Property="MinWidth" Value="100" />
		</Style>
		<Style TargetType="Button">
			<Setter Property="Background" Value="LightCyan" />
			<Setter Property="Foreground" Value="DarkBlue" />
			<Setter Property="Padding" Value="10" />
			<Setter Property="Margin" Value="20" />
			<Setter Property="MinWidth" Value="120" />
			<Setter Property="BorderBrush" Value="Black"/>
		</Style>
	</Page.Resources>

	<Grid Background="DarkCyan">

**Reply by ozdotcom** [reply]
Thanks Carlos,

I have modified to match. looks good.


![image](https://user-images.githubusercontent.com/87345204/227379020-dac3c4f0-9bd2-4911-92d8-59025e4032b7.png)

**Reply by ozdotcom** [reply]
@Jasonlovedataanalyst what's your thoughts on the theme?

Sharryn

**Reply by Jasonlovedataanalyst** [reply]
I will keep up with yours , which looks great

**Reply by ozdotcom** [reply]
Shared the xaml code in the code discussion @Jasonlovedataanalyst

the page resource for style setter is above from Carlos.

**Reply by Jasonlovedataanalyst** [reply]
Thank you for that , I mean for your xaml code for the layout of the mainfraim . I am not quit sure whether yours are done by Grid or Stack panels

**Comment by Jasonlovedataanalyst** [comment]
I have made something similar , I reckon it is should be fine now
![image](https://user-images.githubusercontent.com/87595451/227644812-d3570f16-06d4-43f4-9ac0-a666b20a51c6.png)

**Comment by CarlosPietrasanta** [comment]
Looking good Jason :thumbsup:

Personally I used multiple StackPanels to align everything, but use whatever is easier for you. 

Keep in mind guys this subject is about Git and GitHub, as long as the app merges and works correctly, don't go too crazy on design and layouts. I mean, is great to have a good looking product, but let's not lose focus on the subject.

**Reply by ozdotcom** [reply]
Good point Carlos 

Thanks for keeping us on the right track. 

Sharryn
